spk/syncthing/src/options.conf

```diff
@@ -1,6 +1,9 @@
-#!/bin/sh
-# This file will be added to start-stop-status script
+# This file is included in the start-stop-status script
+# to add additional parameters to start syncthing with.
 # For possible options see syncthing --help
 
-# Example: uncomment this to start syncthing with all devices paused
-# SYNCTHING_OPTIONS="-paused"
+# Example: to start syncthing with all devices paused
+# SYNCTHING_OPTIONS="--paused"
+
+# Do not create a default folder when starting for the first time
+SYNCTHING_OPTIONS="--no-default-folder"
```

Comment:
Good, but I'd rather have this hard-coded in `service-setup.sh` instead.  Making it configurable to the user is mostly useless, because it's only honored on first start anyway.

Reply:
I prefer this solution
- it does not touch existing installations on package update as an update does not overwrite this file
- it is the default for the configurable options on new installations, so it is active on the first start. Afterwards a user may remove this, as it is not honored anymore

Reply:
Exactly my point, it is not honored anymore.  When a user finds this config file and removes the option, they might expect a default folder to magically appear (on next restart?).  But in fact nothing will happen.  If we don't think a default folder makes sense (especially under its default location), let's disable it for good.

Reply:
Besides my thinking that this is a confusing (for the user) way to implement it, technically it won't work either, at least when the wizard is supplied with GUI credentials.  Because `syncthing generate` is called without the `SYNCTHING_OPTIONS` appended, which is the Right Thing to do.  If that was already run, the `--no-default-folder` won't work anymore, because the config was already created during `generate`.

Reply:
When I read the documentation, my understanging was, that the `--no-default-folder` parameter is only evaluated at the first service start (i.e. `syncthing serve` and not with `syncthing generate`), but you are the expert...
If with the parameter `--no-default-folder` something is written to the config, we might adjust the config.xml that is installed with the package and avoid the parameter at all.

Reply:
Oh I forgot we already do ship a `config.xml` as template.  In that case, the option isn't even necessary, as the default folder is *strictly only* added when no XML file exists previously and `--no-default-folder` wasn't passed.  There is nothing to fix here really.

Reply:
In the current config.xml the defaults section looks like this:
```xml
    <defaults>
        <folder ignorePerms="true">
        </folder>
    </defaults>
```

After syncthing is started, there is an attribute `path="~"` added to the folder tag.

For our package the HOME (~) folder is `/var/packages/syncthing/var` - that was the cause for #3320 as this folder is not accessible by users.
Shall we add  `path=""` to config.xml to disable a default path?
With this, at least, there is no default value for the folder, when adding a new folder in syncthing web gui.

The default section would be:
```xml
    <defaults>
        <folder ignorePerms="true" path="">
        </folder>
    </defaults>
```

Reply:
Oh! I think we have a big misunderstanding here.  The XML section you quoted is used as a template when adding a new folder through the GUI or otherwise.  The `path` attribute is what gets suggested in the GUI when adding manually, or when auto-accept is enabled, new folders get created below this path.  It actually makes sense to set this to a more human-friendly (as in accessible via File Station) directory / share.

The other thing we're talking about is a "dummy" folder being created during first run *IF* the XML file does not even exist. That would be a `<folder id="default" ...` element and has nothing to do with the `<defaults><folder ...>` element.  We don't have such a dummy folder in our template config, and it won't get created anyway because we do already install the XML before first starting Syncthing.  Using `--no-default-folder` would make no difference at all.

The best solution IMHO is actually to set `$HOME` to something more sensible.  What options do we have for that?  Only by creating a share from the wizard?

Reply:
The only option I see is to use/create a shared folder a user selects in the install wizard. This is the only supported method to create a folder with read/write access for the internal package user (sc-syncthing).
For all other folders, the users have to give permissions for the package user (with file station)
I can't see that $HOME makes any sense here.

I only want to prevent that a clumsy user takes `/var/packages/syncthing/var` as parent for folders to sync.
So I prefer to add `path=""` to defaults-folder-configuration.

PS: the only misunderstanding was that #3320 proposed to disable the default folder creation.

Reply:
> PS: the misleading item was that #3320 proposed to disable the default folder creation.

Definitely :wink:

Setting `path=""` does work, but `$HOME` is still relevant, as that's what the tilde `~` expands to when entering paths. See this screenshot:
![image](https://user-images.githubusercontent.com/6996887/208208128-16db9870-cd7f-4c5c-a68c-3a10dce48c3b.png)

The same goes for the auto-accept feature. If it gets enabled for a device, the default (template) folder path will be used as a parent for any accepted folders.  So if the only way to have a sensible default location is by asking the user for a new share in the wizard, then I guess that's our best option to avoid troubles.  But we'll need to explain properly what this share is actually used for, and that any other folder can also be synced (with sufficient permissions).